daemon/api_test.go

```diff
@@ -3333,9 +3334,9 @@ func snapList(rawSnaps interface{}) []map[string]interface{} {
 // Tests for GET /v2/interfaces
 
 func (s *apiSuite) TestInterfaces(c *check.C) {
+	builtin.MockInterface(&ifacetest.TestInterface{InterfaceName: "test"})
```

Comment:
Can you please explain this (and similar) changes below. What is the change you are achieving here?

Reply:
This is needed because sanitization code is in all.go at the interfaces level and uses allInterfaces list. 
The mockIface method of the repository only mocks interfaces in the repository, this is not enough for our mock "testiface" interfaces to pass sanitization in these tests.